File: ass5_go/calc/calculate.go
package calc

import (
	"fmt"
    "strconv"
	"ass5_go/converter"
    "ass5_go/operation"
    "ass5_go/stack"
	"ass5_go/valid"
    "ass5_go/queue"
)


func Calculate(query string) float64 {
    result := &queue.QueueUsingSlices{}
    if !valid.IsValid(query) {
        fmt.Println("invalid")
    }
    result = converter.InfixToPostfix(query)
    return Evaluate(result)
    // return 34.65
}

func Evaluate(result *queue.QueueUsingSlices) float64 {

    stk := &stack.StackUsingSlices{}

    for result.Len() != 0 {
        fr := result.Pop().(string)
        if valid.Isnum(rune(fr[0])) {
            num, _ := strconv.ParseFloat(fr, 64)
            stk.Push(num)
        } else {
        val1 := stk.Pop().(float64)
        val2 := stk.Pop().(float64)
        stk.Push(operation.Calc(val2, rune(fr[0]), val1))
        }
    }
    return stk.Peek().(float64)
}

File: Ass5_python/python/include/calculate.py
# import validations as vd, Factory as ft, converter as conv
import validations as vd
import Factory as ft
import converter as conv
import queue

class Calculate(conv.Converter, ft.OperationFactory):
    def __init__(self, query):
        # super().__init__(query)
        obj = conv.Converter(query)
        self.result = queue.Queue()
        self.result = obj.infixToPostfix(query)
        # if e is 1:
        #     print("Please enter valid expression")
    
    def evaluate(self):
        stk = queue.LifoQueue()
        while not self.result.empty():
            fr = self.result.get()
            ec = vd.Validations()
            if ec.isnum(fr[0]):
                num = float(fr)
                stk.put(num)
            else:
                val1 = stk.get()
                val2 = stk.get()
                op = ft.OperationFactory()
                optr = op.factory(fr[0])
                stk.put(optr.calc(val2, val1))

        return stk.get()